onnxruntime/core/framework/allocator.cc

```diff
@@ -15,6 +17,61 @@
 
 #include "core/framework/bfc_arena.h"
 
+using Status = onnxruntime::common::Status;
+
+Status OrtArenaCfg::FromKeyValuePairs(const OrtKeyValuePairs& kvps, OrtArenaCfg& cfg) {
```

Comment:
Support creation from OrtKeyValuePairs to keep the public API minimal. We use OrtKeyValuePairs in general for configuration/options/metadata and this avoids having to add OrtArenaCfg as an additional parameter in some functions like CreateSharedAllocator.